### PR TITLE
Amico

### PR DESCRIPTION
**Pull request by Sara04**
Moving forward model matrix creation to modeling framework; adding distribution estimation with NNLS; adding parameter estimation

### CODE REVIEW

**Comment by rutgerfick** [comment]
hey you have to resolve the conflicts before I can review - you have to rebase you branch on the latest version of master

File: dmipy/optimizers/amico_cvxpy.py [hunk 1]
```diff
@@ -47,103 +47,85 @@ class AmicoCvxpyOptimizer:
...
-
-        # TODO:
-        # M-pruning
-        # estimate x0 vector with non negative least squares
-
         self.x0_vector = x0.value
 
-        return self.x0_vector
+        # 2. Computing distribution vector x0_vector by solving NNLS
+        x0_idx_i = self.x0_vector > x0_th
```
Comment (rutgerfick) [inline dmipy/optimizers/amico_cvxpy.py:105]: daaammmnmnnnnnnnnnn

Comment (rutgerfick) [reply dmipy/optimizers/amico_cvxpy.py:105]: this woman already implemented the selection after the L1

**Comment by rutgerfick** [comment]
Hey @Sara04 I think this solver should be tested on its own.

Can you create a test that only uses this AMICO solver with the required inputs to see if returns a result as expected?

**Comment by Sara04** [comment]
Hey, Rutger, I've added test for amico, but it will evidently require further improvements and investigations

**Comment by rutgerfick** [comment]
hey @Sara04 check out the travis pipeline, your test is not passing and you have some small pep8 issues

**Comment by rutgerfick** [comment]
you still have to fix these guys:

```
./dmipy/optimizers/amico_cvxpy.py:93:16: E127 continuation line over-indented for visual indent
./dmipy/optimizers/amico_cvxpy.py:97:21: E126 continuation line over-indented for hanging indent
./dmipy/optimizers/amico_cvxpy.py:99:21: E127 continuation line over-indented for visual indent
./dmipy/optimizers/tests/test_amico.py:21:1: W293 blank line contains whitespace
./dmipy/optimizers/tests/test_amico.py:51:29: W291 trailing whitespace
./dmipy/optimizers/tests/test_amico.py:55:1: W293 blank line contains whitespace
./dmipy/optimizers/tests/test_amico.py:60:71: W291 trailing whitespace
./dmipy/optimizers/tests/test_amico.py:63:69: W291 trailing whitespace
./dmipy/optimizers/tests/test_amico.py:64:65: W291 trailing whitespace
./dmipy/optimizers/tests/test_amico.py:119:56: E231 missing whitespace after ','
./dmipy/optimizers/tests/test_amico.py:133:1: W293 blank line contains whitespace
./dmipy/optimizers/tests/test_amico.py:140:53: W291 trailing whitespace
```

**Comment by rutgerfick** [comment]
But I notice the amico test is actually not passing (and takes quite long, like 5 min).

Could you please slim this test down to be faster (in the order of seconds, not minutes if possible).

**Comment by rutgerfick** [comment]
you can run the tests locally by just being in your main dmipy folder and typing `nosetests`, or if you just want to run the tests of a single file it is `nosetests path/to/file_test.py`

**Pull request merged**